src/sql/src/plan/typeconv.rs

```diff
@@ -687,13 +634,20 @@ where
             if let ScalarType::Map { .. } = **value_type {
                 unsupported!("nested map types");
             }
-            match plan_cast_to_map(ecx, ccx, expr, &from_typ, &cast_to) {
+            if let ScalarExpr::CallUnary {
+                func: UnaryFunc::CastStringToMap { .. },
+                ..
+            } = expr
+            {
+                return Ok(expr);
+            };
```

Comment:
This special case seems super weird to me!

Reply:
This was me, not Sean!

I was going to write why I needed it, but I think we actually might not anymore. @sploiselle would you mind removing this test case and letting the tests run? I'm also happy to do this in a follow up PR!

Reply:
Should have rested its removal! Looks like it's superfluous now, so all good.

Reply:
Thank you for cleaning up after me!